src/utilities/bcl/BCLMeasure.cpp

```diff
@@ -1012,12 +1014,24 @@ bool BCLMeasure::checkForUpdatesFiles() {
   for (const auto& [subFolderName, usageType] : approvedSubFolderToUsageMap) {
     openstudio::path approvedSubFolderAbsolutePath = m_directory / toPath(subFolderName);
     if (openstudio::filesystem::exists(approvedSubFolderAbsolutePath)) {
-      for (const auto& relativeFilePath : openstudio::filesystem::recursive_directory_files(approvedSubFolderAbsolutePath)) {
-        openstudio::path absoluteFilePath = approvedSubFolderAbsolutePath / relativeFilePath;
-        if (!isApprovedFile(absoluteFilePath, m_directory)) {
-          continue;
+
+      for (auto it = openstudio::filesystem::recursive_directory_iterator(approvedSubFolderAbsolutePath);
+           it != openstudio::filesystem::recursive_directory_iterator(); ++it) {
+        const auto& absoluteFilePath = it->path();
+
+        if (openstudio::filesystem::is_directory(absoluteFilePath)) {
+          const std::string name = absoluteFilePath.filename().string();
+          if (!name.empty() && (name[0] == '.' || name == "__pycache__")) {
+            // Skip this directory and all its children
+            it.disable_recursion_pending();
+          }
+        } else if (openstudio::filesystem::is_regular_file(absoluteFilePath)) {
+          if (!isApprovedFile(absoluteFilePath, m_directory)) {
+            continue;
+          }
+          const openstudio::path relativeFilePath = openstudio::filesystem::relative(absoluteFilePath, m_directory);
+          result |= addWithUsageTypeIfNotExisting(relativeFilePath, std::string(usageType));
```

Comment:
use a recursive_directory_iterator and use `disable_recursion_pending` to skip it and children if it's a hidden folder or named "__pycache__"